superduperdb/components/listener.py

```diff
@@ -83,30 +117,21 @@ def post_create(self, db: Datalayer) -> None:
                 db.cdc.add(self)
 
     @property
-    def dependencies(self) -> t.List[str]:
+    def dependencies(self) -> t.List[ComponentTuple]:
+        args, kwargs = self.mapping.mapping
+        all_ = list(args) + list(kwargs.values())
         out = []
-
-        def check_dependable(k):
-            nonlocal out
-            if k.startswith('_outputs.'):
-                _, key, model = k.split('.')
-                out.append(f'{model}/{key}')
-
-        if isinstance(self.key, str):
-            check_dependable(self.key)
-
-        elif isinstance(self.key, (tuple, list)):
-            for k in self.key:
-                check_dependable(k)
-
-        elif isinstance(self.key, dict):
-            for k in self.key.values():
-                check_dependable(k)
-
-        if self.select.output_fields:
-            out.extend([f'{v}/{k}' for k, v in self.select.output_fields.items()])
+        for x in all_:
+            if x.startswith('_outputs.'):
+                parts = x.split('.')[1].split('::')
+                listener_id, version = '::'.join(parts[:-1]), int(parts[-1])
+                out.append(ComponentTuple('listener', listener_id, version))
         return out
 
+    @property
+    def predict_id(self):
+        return f'{self.identifier}::{self.version}'
```

Comment:
Could we set a predict_id we like? 
If there is no preset predict_id, it will be built automatically and a log will be printed to remind the user.


For example, I build a summary model and add a listener. I just want to mark the output index to be "summary".

Sometimes I don’t care about the intermediate information such as the identifier of the model. I only need to specify its ID when creating the listener, and specify this ID when I get the results.



WDYT?